packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -150,6 +152,7 @@ const MetricForm: FC<MetricFormProps> = ({
   current,
   edit,
   duplicate = false,
+  disabledMessage,
```

Comment:
Like mentioned above, these different forms should not have a `disabledMessage` prop.  Especially in this case, the CTA may be disabled for a couple different reasons.

Instead, I'd expect to see code within this component like this:

```tsx
let ctaEnabled = true;
let disabledMessage = null;

if (riskError) {
  ctaEnabled = false;
  disabledMessage = riskError
} else if (!permissions.check("createDatasources", project)) {
  ctaEnabled = false;
  disabledMessage = "You don't have permission to create data sources";
}

...

<Modal 
  disabledMessage={disabledMessage}
  ctaEnabled={ctaEnabled}
  ...
>
```

If there are other reasons why a CTA might be disabled (like the MetricForm and `riskErrors` for example), we can

Reply:
Yea.... very good point. I've updated all of the affected forms with the pattern above. Thank you!